### PR TITLE
fix(github): fix test conn

### LINKED ISSUE #3075: [Bug][Github] Fail to process invalid token when test connection
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

image
The last token's user is empty, but this is an invalid token

### What you expected to happen

this should show invalid

### How to reproduce

test connection

### Anything else

_No response_

### Version

main

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by warren830**
# Summary

spent a lot time to investigate, finally found that `errors.Default.Wrap` receive a nil for the first param

### Does this close any open issues?
closes #3075

### Screenshots
Include any relevant screenshots here.

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: plugins/github/api/connection.go [hunk 1]
```diff
@@ -90,18 +90,23 @@ func TestConnection(input *core.ApiResourceInput) (*core.ApiResourceOutput, erro
...
 			}
-			results <- VerifyResult{login: githubUserOfToken.Login}
 		}()
 	}
 
 	// collect verification results
 	logins := make([]string, 0)
 	msgs := make([]string, 0)
 	i := 0
+	fmt.Println("444444", len(results))
```
Comment (abeizn) [inline plugins/github/api/connection.go:106]: please delete it.

File: plugins/github/api/connection.go [hunk 2]
```diff
@@ -90,18 +90,23 @@ func TestConnection(input *core.ApiResourceInput) (*core.ApiResourceOutput, erro
...
 	for result := range results {
+		fmt.Println("3333333", result)
```
Comment (abeizn) [inline plugins/github/api/connection.go:108]: same as above

File: plugins/github/api/connection.go [hunk 3]
```diff
@@ -90,18 +90,23 @@ func TestConnection(input *core.ApiResourceInput) (*core.ApiResourceOutput, erro
 				results <- VerifyResult{err: errors.Default.Wrap(err, fmt.Sprintf("verify token failed for #%v %s", j, token))}
 				return
 			} else if githubUserOfToken.Login == "" {
-				results <- VerifyResult{err: errors.Default.Wrap(err, fmt.Sprintf("invalid token for #%v %s", j, token))}
+				results <- VerifyResult{err: errors.Default.Wrap(errors.Unauthorized.New("invalid token"), fmt.Sprintf("invalid token for #%v %s", j, token))}
```
Comment (klesh) [inline plugins/github/api/connection.go:93]: Please remove the token from the error message. We don't need it anymore

File: plugins/github/api/connection.go (hunk 1 above)
Comment (warren830) [reply plugins/github/api/connection.go:106]: > please delete it.

Hahaha, I was crashed

**Comment by warren830** [comment]
This is fixed by @keon94 's pr #2984 
But imported another bug:
image

**Pull request closed without merging**